### ISSUE TITLE
Add support for NaN equality within Arrays

### ISSUE THREAD

**Issue by mitches-got-glitches** [MrPowers/chispa#27]
When trying to `assert_df_equality` with `allow_nan_equality=True`, if the both DataFrames hold an array that contains some `nan` values then the comparer fails, even if the `nan`s are in the same place.

An additional check should take place here to compare the array elements. 

https://github.com/MrPowers/chispa/blob/500793efe14b1975b86fb1a923ee6cd68ba559d8/chispa/row_comparer.py#L18

Again, I can help contribute to this one but it won't be until next week.

Here's a comparison example:

![image](https://user-images.githubusercontent.com/42358474/115743341-80b31880-a389-11eb-96bd-a087d0ab7ae6.png)

**Comment by MrPowers** [comment]
Yes, this would be a good addition, and no rush, whenever you get around to it!  Thanks so much for the help!!

**Comment by mitches-got-glitches** [comment]
Hi @MrPowers , just a note to say I have a pull request on this sitting ready to go. I branched off the changes on  #29 though because it made sense to get that one through first. I'm just wondering what your timeline might be for a further review of the impact of #29? Some of the features would be useful to me currently.
